plugins/backstage-plugin-flux/src/hooks/useToggleSuspendResource.ts

```diff
@@ -0,0 +1,153 @@
+import { AlertApi, alertApiRef, useApi } from '@backstage/core-plugin-api';
+import { KubernetesApi, kubernetesApiRef } from '@backstage/plugin-kubernetes';
+import { CustomResourceMatcher } from '@backstage/plugin-kubernetes-common';
+import { useAsyncFn } from 'react-use';
+import { gvkFromKind } from '../objects';
+import { Deployment, Source } from '../components/helpers';
+
+export const pathForResource = (
+  name: string,
+  namespace: string,
+  gvk: CustomResourceMatcher,
+): string => {
+  const basePath = [
+    '/apis',
+    gvk.group,
+    gvk.apiVersion,
+    'namespaces',
+    namespace,
+    gvk.plural,
+    name,
+  ].join('/');
+
+  return basePath;
+};
+
+export function toggleSuspendRequest(
+  name: string,
+  namespace: string,
+  clusterName: string,
+  gvk: CustomResourceMatcher,
+  suspend: boolean,
+) {
+  return {
+    clusterName,
+    path: pathForResource(name, namespace, gvk),
+    init: {
+      method: 'PATCH',
+      headers: {
+        'Content-Type': 'application/merge-patch+json',
+      },
+      body: JSON.stringify({
+        spec: {
+          suspend,
+        },
+      }),
+    },
+  };
+}
+
+export function getRequest(
+  name: string,
+  namespace: string,
+  clusterName: string,
+  gvk: CustomResourceMatcher,
+) {
+  return {
+    clusterName,
+    path: pathForResource(name, namespace, gvk),
+  };
+}
+
+export async function requestToggleSuspendResource(
+  kubernetesApi: KubernetesApi,
+  name: string,
+  namespace: string,
+  clusterName: string,
+  gvk: CustomResourceMatcher,
+  suspend: boolean,
+) {
+  const res = await kubernetesApi.proxy(
+    toggleSuspendRequest(name, namespace, clusterName, gvk, suspend),
+  );
+  const key = suspend ? 'Suspend' : 'Resume';
+  if (!res.ok) {
+    throw new Error(
+      `Failed to ${key} resource: ${res.status} ${res.statusText}`,
+    );
+  }
+}
+
+export async function toggleSuspendResource(
+  resource: Source | Deployment,
+  kubernetesApi: KubernetesApi,
+  alertApi: AlertApi,
+  suspend: boolean,
+) {
+  const key = suspend ? 'Suspend' : 'Resume';
+  try {
+    const gvk = gvkFromKind(resource.type);
+    if (!gvk) {
+      throw new Error(`Unknown resource type: ${resource.type}`);
+    }
+
+    if ('sourceRef' in resource && resource.sourceRef) {
+      const sourceGVK = gvkFromKind(resource.sourceRef.kind);
+      if (!sourceGVK) {
+        throw new Error(
+          `Unknown resource source type: ${resource.sourceRef.kind}`,
+        );
+      }
+      // sync the source
+      await requestToggleSuspendResource(
+        kubernetesApi,
+        resource.sourceRef.name!,
+        resource.sourceRef.namespace || resource.namespace,
+        resource.clusterName,
+        sourceGVK,
+        suspend,
+      );
+    }
```

Comment:
I don't think we want to resume / suspend the source as well as the ks/hr in this case.

Other ks/hr could be relying on the same source and they would stop updating too which might be susprising.

E.g. The `flux` cli only provides the option to update the source for `reconcile` (sync)

```
❯ flux resume kustomization --help | grep with
❯ flux suspend kustomization --help | grep with
❯ flux reconcile kustomization --help | grep with
  flux reconcile kustomization podinfo --with-source
      --with-source   reconcile Kustomization source
```

Reply:
Yes, we should only suspend the viewed resource.